modules/internal/ChapelBase.chpl

```diff
@@ -1293,12 +1311,15 @@ module ChapelBase {
   }
 
   inline proc %=(ref lhs:int(?w), rhs:int(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0 then
+        halt("Attempt to compute a modulus by zero");
     __primitive("%=", lhs, rhs);
   }
   inline proc %=(ref lhs:uint(?w), rhs:uint(w)) {
-    __primitive("%=", lhs, rhs);
-  }
-  inline proc %=(ref lhs:real(?w), rhs:real(w)) {
```

Comment:
Unless I'm reading the diff wrong, it appears that you've completely removed the %= overload for reals, which I don't think we want to do.